### ISSUE TITLE
Date / Datetime Field: flatpickr two-way data binding problem

### ISSUE THREAD

**Issue by davi5e** [laravel/nova-issues#2344]
- Laravel Version: 6.15.0
- Nova Version:  v2.10.1 
- PHP Version: 7.3.14
- Database Driver & Version:
- Operating System and Version: Docker php:7.3-fpm-alpine
- Browser type and version: Mozilla Firefox for Ubuntu 72.0.2 64-bit

### Description:

After using the datetime picker to fill a nullable [Date](https://nova.laravel.com/docs/2.0/resources/fields.html#date-field) or [DateTime](https://nova.laravel.com/docs/2.0/resources/fields.html#datetime-field) field there is no way to reset it, as in clear the field setting the value back to null.

This doesn't work very well with Laravel's validation code. Seems to me like a problem with Vue's two-way binding system or a lack of `$emit` to tell some other component the value was set directly into the input, **without** using flatpickr. This part is pure guessing though.

### Steps To Reproduce:

1. Create a Date or DateTime field, mark it `nullable()`
2. Set rules, like `rules('nullable', 'date', 'after:today')`
3. Select some **invalid** date (_today_ works for the above rules)
4. Delete all charactes from the field
5. Send the form, see the validation error being displayed.

I'd just add a clear button, seems to solve the problem. [Here's an example](https://flatpickr.js.org/examples/#flatpickr-external-elements) from flatpickr's doc.

As a way to make it less complex, I think that disabling the direct input would eliminate part of the problem since it'd force the user to always use flatpickr. The clear button is still needed in this scenario.

### Example Code

```php
    Date::make('Valid Until')
                ->withMeta([
                    'placeholder' => 'Choose a date or leave it blank...',
                ])
                ->rules('nullable', 'date', 'after:today')
                ->nullable(),
```

**Comment by ziming** [comment]
Agree, have this problem as well. If I clear the datetime field value and click somewhere else, the date time field gets filled with a value again

**Comment by JulianMar** [comment]
Would be really cool to have this feature.
It was possible to clear the value before Nova 2.9 I think

**Comment by florentpoujol** [comment]
I have found a way to clear the field :
- empty the field manually
- press `Enter` to unfocus the field, the picker is still here, but the field is not highlighted anymore
- then press `Escape` to make the picker go away, and the field will stay empty

@davidhemphill It would still be cool if there was a more practical way of doing this.

**Comment by jbrooksuk** [comment]
Duplicate of https://github.com/laravel/nova-issues/issues/2358

**Comment by zachgarcia1** [comment]
Is this going to get fixed anytime soon? it seems to have been an issue for a while now. I know its a duplicate of #2358 but that issue has been locked

**Comment by davidhemphill** [comment]
@zachgarcia1 This was fixed via laravel/nova#907, which shows a "clear" button when the field is set with the `nullable` modifier.

**Comment by immediate43** [comment]
> Agree, have this problem as well. If I clear the datetime field value and click somewhere else, the date time field gets filled with a value again





https://tracey582.github.io/e0XWIEjp89Sxn1i3W1tlxBciFRbq8wliZXC1xi06uukUEeVslMkvQZh0/
